src/main/java/org/zalando/nakadi/controller/EventStreamController.java

```diff
@@ -181,6 +190,7 @@ public StreamingResponseBody streamEvents(
                 client.checkScopes(eventType.getReadScopes());
 
                 // validate parameters
+                final TopicRepository topicRepository = timelineService.getTopicRepository(eventType);
```

Comment:
Shouldn't topic be retrieved based on `event type + cursor`? I mean, even if you don't plan to implement it now, since this is a refactoring, maybe it would be a good idea to define an API closer to the final version, something like `topicRepositoryPool.get(eventType, cursor)` that returns something like a map from cursor to repository. Wdyt?